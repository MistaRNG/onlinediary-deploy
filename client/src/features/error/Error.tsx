import { Toaster, Toast } from "@blueprintjs/core";
import useError from "common/hooks/useError";
import React from "react";

const Error: React.FC = () => {
  const { displayedError, dismissHandler } = useError();

  return (
    displayedError && (
      <Toaster>
        <Toast
          onDismiss={dismissHandler}
          message={displayedError}
          intent="danger"
        />
      </Toaster>
    )
  );
};

export default Error;
