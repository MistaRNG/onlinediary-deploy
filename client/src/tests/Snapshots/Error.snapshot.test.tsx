import React from 'react';
import { render } from '@testing-library/react';
import Error from '../../features/error/Error';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import errorReducer from '../../features/error/errorSlice';
import { describe, it, expect } from '@jest/globals';

const mockStore = configureStore({
  reducer: {
    error: errorReducer,
  },
});

describe('Error Component Snapshot Tests', () => {
  it('should render the Error component correctly', () => {
    // ARRANGE & ACT
    const { asFragment } = render(
      <Provider store={mockStore}>
        <Error />
      </Provider>
    );

    // ASSERT
    expect(asFragment()).toMatchSnapshot();
  });
});
