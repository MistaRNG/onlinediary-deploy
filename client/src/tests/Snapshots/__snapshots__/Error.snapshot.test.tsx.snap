// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Component Snapshot Tests should render the Error component correctly 1`] = `<DocumentFragment />`;
